src/strategies/StrategyALL.java

```diff
@@ -0,0 +1,32 @@
+package strategies;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+public class StrategyALL implements SearchStrategy {
```

Comment:
Can you create a folder `impl` and keep all of your implementations there?
Also, please rename your classes, e.g. `StrategyALL` -> `AllStrategyImpl`

Reply:
done

Reply:
Left a small comment about package structure in another place.